services/text-formatters.js

```diff
@@ -46,27 +46,31 @@ function ordinalNumber(n) {
   return n + (s[(v - 20) % 10] || s[v] || s[0])
 }
 
-// Given a number, string with appropriate unit in the metric system, SI.
+// Given a number (positive or negative), string with appropriate unit in the metric system, SI.
```

Comment:
My initial thought was that we'd introduce a new function, and thus formatter, that would support positive/negative while leveraging `metric` internally. In hindsight though I'm questioning myself and whether that'd actually be helpful, and think I'm convinced by having a proverbial one-stop-shop as you've done here.

However, I do think we should increase the unit test suite (can be found in `text-formatters.spec.js`) to account for the newly supported inputs

Reply:
I thought about this too. The conclusion I came to before writing https://github.com/badges/shields/pull/7411#discussion_r774163739 is I don't see any downside to `metric()` just working for both positive and negative as long as we've got separate test validators for metric (positive only) and metric (including zero/negative).

We do need some tests in https://github.com/badges/shields/blob/23c0406bedfc6930735e8f5ea75dfe34faf1f290/services/text-formatters.spec.js#L40 for the negative cases though :+1:

Reply:
Gotcha, sorry not sure how I missed that. I blame the eggnog and/or holidays in general :laughing: :gift: